### DISCUSSION TITLE
Load BingMaps before stories

### DISCUSSION THREAD

**Discussion by rexkenley** [storybookjs/storybook#15727] [Help]
I am creating a ui component using BingMaps & storybook. My issue is that BingMaps "script" creates a "Microsoft" object in the global scope. However, this object gets loaded after the stories. Is there a way to make sure that the Microsoft object gets created first before the stories get processed?

I have tried preview-head.html and previewHead in main.js to no avail.

![image](https://user-images.githubusercontent.com/10548251/127703167-cfcca537-78ee-43f6-8cf6-20e8f6345c6d.png)
 
[Creating BingMaps Control](https://docs.microsoft.com/en-us/bingmaps/v8-web-control/creating-and-hosting-map-controls/creating-a-basic-map-control)

**Comment by shilman** [comment]
I would expect `preview-head`/`previewHead` would do the trick. Have you tried debugging?

If you need it, a more heavy-handed approach could be loaders: https://storybook.js.org/docs/react/writing-stories/loaders

You could maybe write a loader that returns nothing but simply waits for the maps to be loaded.

**Reply by rexkenley** [reply]
Yes, as shown on the screen shot, the Microsoft object is undefined right before it is being called.

I will take a look at the doc, thanks!
